### PR TITLE
E2EE. Fix root metadata fetching path for non-root remote sync folder. Refactoring. Stabilizing paths.

### PR DESCRIPTION
**Pull request by allexzander**

### CODE REVIEW

**Comment by allexzander** [comment]
Yes. Again, lots of similar changes for dealing with non-root remote sync root paths. Due to the new way of fetching metadata (keys are stored in the root folder's metadata). Extensive manual tests were successful. What we are really missing is end-to-end testing. Otherwise, we are going to continue fixing regressions.

File: src/libsync/vfs/cfapi/hydrationjob.h [hunk 1]
```diff
@@ -45,9 +45,9 @@ class HydrationJob : public QObject
 
     AccountPtr account() const;
     void setAccount(const AccountPtr &account);
-
-    QString remotePath() const;
-    void setRemotePath(const QString &remotePath);
+
```
Comment (mgallien) [inline src/libsync/vfs/cfapi/hydrationjob.h:48]: please remove the extra spaces

**Review by mgallien — CHANGES_REQUESTED** [review]

**Review by mgallien — CHANGES_REQUESTED** [review]
please add automated tests for such cases
would be needed to ensure this code can become more reliable

**Comment by allexzander** [comment]
> please add automated tests for such cases would be needed to ensure this code can become more reliable

How do you propose to implement these tests? The e2e-to-end encryption interaction between client and server is way too complicated to implement unit tests for it. This requires writing most of the endpoints the server is using for E2EE, not just simple PROPFIND. We do not have framework for such testing for this reason.

This type of tests require different testing approach, called end-to-end testing (or QA automation), for which, we do not have capacity.

**Review by mgallien — APPROVED** [review]

**Pull request merged**